### PR TITLE
[FAB-17993] Skip modifying _users security object

### PR DESCRIPTION
**Pull request by lindluni**
Introduces a check to skip modifying the security object of the `_users` table. The latest versions of CouchDB protect from this operation being performed and it isn't necessary for the peer to actually modify the object to perform its functionality

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

### CODE REVIEW

**Comment by denyeart** [comment]
LGTM and this does simplify it somewhat, but would also like @manish-sethi or @cendhu to look before merging though.

File: core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go [hunk 1]
```diff
@@ -232,73 +231,46 @@ func (dbclient *couchDatabase) createDatabaseIfNotExist() error {
...
-	defer closeResponseBody(resp)
 
-	errSecurity := dbclient.applyDatabasePermissions()
-	if errSecurity != nil {
-		return errSecurity
+	if dbclient.dbName != "_users" {
+		errSecurity := dbclient.applyDatabasePermissions()
+		if errSecurity != nil {
+			return errSecurity
+		}
```
Comment (cendhu) [inline core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go:269]: Can't we just move the dbName check to the `applyDatabasePermissions()` itself? This ensures that this concept of applying permission to non `_user` table is kept close to each other.

Comment (cendhu) [reply core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go:269]: This will also help in returning early as per line 262 if db exists already?

Comment (lindluni) [reply core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go:269]: It's generally bad practice for the first check of a function to be a singular condition on which the function simply returns without an error. Its like passing a value that could potentially be nil into a function, it's the caller's responsibility to check the state of the objects being invoked on, not the callee.

Comment (manish-sethi) [reply core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go:269]: I am fine with this as is.

**Review by manish-sethi — APPROVED** [review]

File: core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go (hunk 1 above)
Comment (cendhu) [reply core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go:269]: I see. Thanks, Brett. Make sense for the caller to do it instead of callee.

**Comment by btl5037** [comment]
@Mergifyio backport release-2.1

**Pull request merged**